src/status_im/contexts/keycard/migrate/events.cljs

```diff
@@ -0,0 +1,125 @@
+(ns status-im.contexts.keycard.migrate.events
+  (:require [clojure.string :as string]
+            [status-im.contexts.keycard.pin.create.view :as pin.create]
+            [status-im.contexts.keycard.pin.view :as keycard.pin]
+            [utils.re-frame :as rf]
+            [utils.security.core :as security]))
+
+(rf/reg-event-fx :keycard/migration.check-empty-card
+ (fn [{:keys [db]}]
+   {:fx [[:dispatch
+          [:keycard/connect
+           {:key-uid (get-in db [:profile/profile :key-uid])
+            :on-success
+            (fn []
+              ;;TODO keys already on the keycard, new flow needs to be implemented
+              ;; https://github.com/status-im/status-mobile/issues/21446
+              (rf/dispatch [:keycard/disconnect])
+              (rf/dispatch [:open-modal :screen/keycard.authorise
+                            {:on-success
+                             #(rf/dispatch [:keycard/migration.authorisation-success %])}]))
+            :on-error
+            (fn [error]
+              (if (= error :keycard/error.keycard-blank)
+                (do
+                  (rf/dispatch [:keycard/disconnect])
+                  (rf/dispatch [:open-modal :screen/keycard.empty]))
+                (rf/dispatch [:keycard/on-application-info-error error])))}]]]}))
+
+(defn get-application-info-and-continue
+  [key-uid]
+  (rf/dispatch [:keycard/get-application-info
+                {:key-uid key-uid
+                 :on-success #(rf/dispatch [:keycard/migration.continue])
+                 :on-error
```

Comment:
Maybe a nice one for future PRs: it's a bit confusing that sometimes we use `:on-error` and sometimes `:on-failure` (e.g. in `keycard.utils/wrap-handlers`).

Reply:
yes it's confusing for sure, i've asked this question if discord, we need a guideline for this

Reply:
https://github.com/status-im/status-mobile/issues/21511